mixer/pkg/protobuf/yaml/dynamic/builder.go

```diff
@@ -0,0 +1,280 @@
+				ma, ok = v.([]interface{})
+				if !ok {
+					return nil, fmt.Errorf("unable to process: %v, got %T, want: []interface{}", fd, v)
+				}
+			} else {
+				ma = []interface{}{v}
+			}
+
+			// now maps, messages and repeated messages all look the same.
+			if err = c.buildMessageField(ma, m, fd, &me); err != nil {
+				return nil, err
+			}
+
+		default:
+			return nil, fmt.Errorf("field not supported '%v'", fd)
+		}
+	}
+
+	// sorting fields is recommended
+	sort.Slice(me.fields, func(i, j int) bool {
+		return me.fields[i].number < me.fields[j].number
+	})
+
+	return me, nil
+}
+
+// buildPrimitiveField handles encoding of single and repeated packed primitives
+func (c Builder) buildPrimitiveField(v interface{}, fd *descriptor.FieldDescriptorProto,
+	static bool) (*field, error) {
+	fld := makeField(fd)
+	e := c.resolver.ResolveEnum(fd.GetTypeName())
+
+	va, packed := v.([]interface{})
+	if !packed {
+		va = []interface{}{v}
+	}
+
+	for _, vl := range va {
+		var err error
+
+		val, isConstString := transFormQuotedString(vl)
+		sval, isString := val.(string)
+
+		var enc Encoder
+
+		// if compiler is nil, everything is considered static
+		if static || isConstString || !isString || c.compiler == nil {
+
+			if fd.IsEnum() {
+				pe := &staticEncoder{name: fd.GetName()}
+				pe.encodedData, err = EncodeEnum(val, pe.encodedData, e.Value)
+				enc = pe
+			} else {
+				enc, err = BuildPrimitiveEncoder(val, fd)
+			}
+
+			if err != nil {
+				return nil, fmt.Errorf("unable to build primitive encoder: %v. %v", val, err)
+			}
+		} else {
+			var expr compiled.Expression
+			var vt v1beta1.ValueType
+
+			if expr, vt, err = c.compiler.Compile(sval); err != nil {
+				return nil, err
+			}
+
+			if fd.IsEnum() {
+				// enums can be expressed as String or INT only
+				if !(vt == v1beta1.INT64 || vt == v1beta1.STRING) {
+					return nil, fmt.Errorf("unable to build eval encoder: enum %v for expression type:%v", val, vt)
+				}
+				enc = &eEnumEncoder{
+					expr:       expr,
+					enumValues: e.Value,
+				}
+			} else {
+				enc, err = BuildPrimitiveEvalEncoder(expr, vt, fd)
+			}
+
+			if err != nil {
+				return nil, fmt.Errorf("unable to build eval encoder: %v. %v", val, err)
+			}
+		}
+
+		// ASSERT(enc != nil)
+		fld.encoder = append(fld.encoder, enc)
+	}
+
+	return fld, nil
+}
+
+func (c Builder) buildMessageField(ma []interface{}, m *descriptor.DescriptorProto,
+	fd *descriptor.FieldDescriptorProto, me *messageEncoder) error {
+
+	for _, vv := range ma {
+		var ok bool
+
+		var vq map[interface{}]interface{}
+		if vq, ok = vv.(map[interface{}]interface{}); !ok {
+			return fmt.Errorf("unable to process: %v, got %T, want: map[string]interface{}", fd, vv)
+		}
+
+		var de Encoder
+		var err error
+		if de, err = c.buildMessage(m, vq, false); err != nil {
+			return fmt.Errorf("unable to process: %v, %v", fd, err)
+		}
+		fld := makeField(fd)
+		fld.encoder = []Encoder{de}
+		me.fields = append(me.fields, fld)
```

Comment:
why is this multiple fields as oppose to multiple encoders within the same field